app/src/main/kotlin/at/bitfire/davdroid/sync/SyncManager.kt

```diff
@@ -727,6 +743,20 @@ abstract class SyncManager<ResourceType: LocalResource<*>, out CollectionType: L
         return state
     }
 
+    /**
+     * Adds `Push-Dont-Notify` header to the request if a push subscription exists and has not expired yet.
+     * @return a map containing the header if a push subscription is active, otherwise an empty map.
+     */
+    private fun pushDontNotifyHeader(): Map<String, String> {
+        val expired = collection.pushSubscriptionExpires?.let { expires ->
+            val now = (Instant.now() + Duration.ofMinutes(1)).epochSecond
+            now >= expires
+        } == true
+        return if (!expired && collection.pushSubscription != null)
+            mapOf("Push-Dont-Notify" to collection.pushSubscription)
+        else
+            emptyMap()
+    }
```

Comment:
It's calculated between zero (no uploads) and many (many uploads) times with always the same result*.

Shall we put it into a `val by lazy`?

* Despite from the `expires`. Guess I was wrong when I said we should check for expiration here. That's the task of `PushRegistrationManager`, which runs regularly.

Reply:
I think we can even just extract it to collection as a computed property.